src/libpsl-native/src/issamefile.cpp

```diff
@@ -0,0 +1,47 @@
+//! @file isdirectory.cpp
+//! @author Andrew Schwartzmeyer <andschwa@microsoft.com>
+//! @brief returns if the path is a directory
+
+#include "getstat.h"
+#include "issamefile.h"
+
+#include <assert.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+//! @brief returns if two paths refer to the same file; uses stat and so follows symlinks
+//!
+//! IsSameFile
+//!
+//! @param[in] path_one
+//! @parblock
+//! A pointer to the buffer that contains the first path.
+//!
+//! char* is marshaled as an LPStr, which on Linux is UTF-8.
+//! @endparblock
+//!
+//! @param[in] path_one
+//! @parblock
+//! A pointer to the buffer that contains the second path.
+//!
+//! char* is marshaled as an LPStr, which on Linux is UTF-8.
+//! @endparblock
+//!
+//! @retval true if directory, false otherwise
+//!
+bool IsSameFile(const char* path_one, const char* path_two)
```

Comment:
I agree that this could use a better name, since it does work for both files and directories, but I'm not sure that `IsSamePath` is the best choice. I avoided `IsSamePath` because two different paths can arrive at the same destination. I went with `IsSameFile` just because the docs for the `stat` function and the `GetFileInformationByHandle` both just use the term "file" even though they both work for files and directories.